static/app/components/events/interfaces/performance/spanEvidenceKeyValueList.tsx

```diff
@@ -219,6 +225,51 @@ function getSpanEvidenceValue(span: Span | null): string {
   return `${span.op} - ${span.description}`;
 }
 
+const getConsecutiveDbTimeSaved = (
+  consecutiveSpans: Span[],
+  independentSpans: Span[]
+): number => {
+  const totalDuration = sumSpanDurations(consecutiveSpans);
+  const maxIndependentSpanDuration = Math.max(
+    ...independentSpans.map(span => getSpanDuration(span))
+  );
+  const independentSpanIds = independentSpans.map(span => span.span_id);
+
+  let sumOfDependentSpansDuration = 0;
+  consecutiveSpans.forEach(span => {
+    if (!independentSpanIds.includes(span.span_id)) {
+      sumOfDependentSpansDuration += getSpanDuration(span);
+    }
+  });
+
+  return (
+    totalDuration - Math.max(maxIndependentSpanDuration, sumOfDependentSpansDuration)
+  );
+};
+
+const sumSpanDurations = (spans: Span[]) => {
+  let totalDuration = 0;
+  spans.forEach(span => {
+    totalDuration += getSpanDuration(span);
+  });
+  return totalDuration;
+};
+
+const getSpanDuration = ({timestamp, start_timestamp}: Span) => {
+  return timestamp && start_timestamp ? (timestamp - start_timestamp) * 1000 : 0;
+};
+
+function getDurationImpact(event: EventTransaction, durationAdded: number) {
+  const transactionTime = (event.endTimestamp - event.startTimestamp) * 1000;
+  if (!transactionTime) {
+    return null;
+  }
+  const percent = durationAdded / transactionTime;
+  return `${toPercent(percent)} (${getPerformanceDuration(
+    durationAdded
+  )}/${getPerformanceDuration(transactionTime)})`;
+}
+
```

Comment:
I think you should refactor this and `getsingleSpanDurationImpact` so that the same code isn't repeated twice. Have `getSingleSpanDurationImpact` take the span and get its duration before returning `getDurationImpact(event, spanDuration)`.

Reply:
make sense! I'll make the change.

Reply:
Made the change!